pkg/k8sapi/k8sutils.go

```diff
@@ -0,0 +1,77 @@
+package k8sapi
+
+import (
+	"fmt"
+	eniconfigscheme "github.com/aws/amazon-vpc-cni-k8s/pkg/apis/crd/v1alpha1"
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/utils/logger"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var log = logger.Get()
+
+// CreateKubeClient creates a k8s client
+func CreateKubeClients() (client.Client, client.Client, error) {
+	restCfg := ctrl.GetConfigOrDie()
+	vpcCniScheme := runtime.NewScheme()
+	clientgoscheme.AddToScheme(vpcCniScheme)
+	eniconfigscheme.AddToScheme(vpcCniScheme)
+
+	stopChan := ctrl.SetupSignalHandler()
+	cache, err := cache.New(restCfg, cache.Options{Scheme: vpcCniScheme})
+	if err != nil {
+		return nil, nil, err
+	}
+	go func() {
+		cache.Start(stopChan)
+	}()
+	cache.WaitForCacheSync(stopChan)
+
+	standaloneK8SClient, err := client.New(restCfg, client.Options{Scheme: vpcCniScheme})
+	k8sClient := client.DelegatingClient{
```

Comment:
Afaics the only difference here is the read cache?  Perhaps we should call them 'caching' and 'raw' clients rather than 'standalone' and .. 'not-standalone' ?

Reply:
Yes, read cache. Agree on the naming, will adjust the names..